presto-main/src/main/java/io/prestosql/sql/planner/LocalDynamicFilterConsumer.java

```diff
@@ -111,12 +121,23 @@ private synchronized void addPartition(TupleDomain<String> tupleDomain)
         return builder.build();
     }
 
-    public static Optional<LocalDynamicFilter> create(JoinNode planNode, TypeProvider types, int partitionCount)
+    private Map<String, Domain> convertTupleDomain(TupleDomain<String> result)
+    {
+        if (result.isNone()) {
+            // One of the join build symbols has no non-null values, therefore no filters can match predicate
+            return buildChannels.keySet().stream()
+                    .collect(toImmutableMap(identity(), filterId -> Domain.none(filterBuildTypes.get(filterId))));
+        }
+        // Compact TupleDomain before reporting dynamic filters to coordinator through TaskStatus to avoid bloating QueryInfo
```

Comment:
this isn't the right place since some connectors might take advantage of larger filter. This should be done in https://github.com/prestosql/presto/pull/1072/commits/1a2665394425e75ff54d6c5c028883d226f0f7c4#diff-fe87ad26e5ec4a22a8bc522e5b9e2fbeR277

Reply:
I can make this change, but I didn't get how moving the `simplify` call to SqlTask would allow more flexibility for connectors. At what point in the flow of reporting dynamic filters to coordinator would we be able to plug in connector specific logic ? By the time `ConnectorSplitManager#getSplits` is called in coordinator, the domains would have been compacted in either case.
Also, doing it here has a small advantage that we call `simplify` only once for a TupleDomain.

Reply:
> I can make this change, but I didn't get how moving the simplify call to SqlTask would allow more flexibility for connectors

dynamic filters collected here are used for:
1. local dynamic filtering (`io.prestosql.split.PageSourceProvider#createPageSource`)
2. passing dynamic filter data back to coordinator

If we simplify here then for 1) connectors will always get simplified dynamic filter. For example, in Hive connector simplification happens in `io/prestosql/plugin/hive/HivePageSourceProvider.java:117`

Reply:
For local dynamic filtering we're using `convertTupleDomainForLocalFilters` where we don't simplify TupleDomain. `convertTupleDomain` is used only when reporting filters to coordinator.

Reply:
Its not obvious from the code that it's the case. There two methods should behave similarly to avoid confusion.

Reply:
Ok, I've now moved the TupleDomain compaction to SqlTask.